lib/shopify_cli/commands/login.rb

```diff
@@ -33,17 +31,25 @@ def call(*)
           IdentityAuth.new(ctx: @ctx).authenticate
           org = select_organization
           ShopifyCLI::DB.set(organization_id: org["id"].to_i) unless org.nil?
-          Whoami.call([], "whoami")
+
         end
+        # validate that shop belongs to organization
+        ShopifyCLI::DB.set(shop: self.class.validate_shop(shop, org, context: @ctx)) unless shop.nil?
+        Whoami.call([], "whoami")
       end
 
       def self.help
         ShopifyCLI::Context.message("core.login.help", ShopifyCLI::TOOL_NAME)
       end
 
-      def self.validate_shop(shop, context:)
+      def self.validate_shop(shop, org, context:)
```

Comment:
Nitpick: I'd use keyed arguments to make it clear from the caller what every argument means.
```rb
def self.validate_shop(shop:, org:, context:)
```